### PR TITLE
Ports TG seethrough components

### PR DESCRIPTION
**Pull request by MisterGrimm**
## About The Pull Request
Ports the seethrough components for objects and mobs, allowing you to see through big objects you are behind, and hit mobs hiding behind you when you are a big mob.

Seethrough objects made by @Time-Green on https://github.com/tgstation/tgstation/pull/69642 , and seethrough mobs made by @IndieanaJones on https://github.com/tgstation/tgstation/pull/77361 .

## Why It's Good For The Game
Lets you see and hit people hiding behind large objects like trees, and also allows big player controlled mobs to hit mobs hiding behind their sprite.

## Changelog

:cl: Time-Green, IndieanaJones
qol: Adds the seethrough component, allowing people to see behind large objects, and behind their own mob if they are large
/:cl:

### CODE REVIEW

**Comment by Kapu1178** [comment]
PR needs creditting original PR/authors.

**Review by Kapu1178 — CHANGES_REQUESTED** [review]
zcopy.dm line 614

put this
```
GLOBAL_REAL_VAR(zmimic_fixed_planes) = list(
	"0" = "World plane (Non-Z)",
	STRINGIFY(GAME_PLANE) = "Game plane (Non-Z)",
	STRINGIFY(FLOOR_PLANE) = "Floor plane (Non-Z)",
	STRINGIFY(GRAVITY_PULSE_PLANE) = "Gravity pulse plane (Non-Z)",
	STRINGIFY(HEAT_PLANE) = "Heat plane (Non-Z)"
)
```

**Comment by MisterGrimm** [comment]
> zcopy.dm line 614
> 
> put this
> 
> ```
> GLOBAL_REAL_VAR(zmimic_fixed_planes) = list(
> 	"0" = "World plane (Non-Z)",
> 	STRINGIFY(GAME_PLANE) = "Game plane (Non-Z)",
> 	STRINGIFY(FLOOR_PLANE) = "Floor plane (Non-Z)",
> 	STRINGIFY(GRAVITY_PULSE_PLANE) = "Gravity pulse plane (Non-Z)",
> 	STRINGIFY(HEAT_PLANE) = "Heat plane (Non-Z)"
> )
> ```

Done.

File: code/__DEFINES/layers.dm [hunk 1]
```diff
@@ -22,7 +22,10 @@ Specifically: ZMIMIC_MAX_PLANE to (ZMIMIC_MAX_PLANE - ZMIMIC_MAX_DEPTH)
 #define GRAVITY_PULSE_RENDER_TARGET "*GRAVPULSE_RENDER_TARGET"
 
 
-#define FLOOR_PLANE -7
+#define FLOOR_PLANE -8
+
+///Slightly above the game plane but does not catch mouse clicks. Useful for certain visuals that should be clicked through, like seethrough trees
+#define SEETHROUGH_PLANE -7
```
Comment (Kapu1178) [inline code/__DEFINES/layers.dm:28]: This is not above the game plane, this will draw below it.

**Review by Kapu1178 — CHANGES_REQUESTED** [review]

File: code/__DEFINES/layers.dm (hunk 1 above)
Comment (MisterGrimm) [reply code/__DEFINES/layers.dm]: Fixed.

**Review by Kapu1178 — APPROVED** [review]

**Pull request merged**